### PR TITLE
Detect new line style in init.

### PR DESCRIPTION
**Pull request by gzzz**
https://github.com/floridoo/gulp-sourcemaps/pull/235#issuecomment-255593650

### CODE REVIEW

File: src/init.js [hunk 1]
```diff
@@ -41,6 +42,8 @@ function init(options) {
     var fileContent = file.contents.toString();
     var sourceMap;
 
+    file.newline = detectNewline.graceful(fileContent || '');
```
Comment (nmccready) [inline src/init.js]: Good idea,  I am working on some other workarounds by tackingon fields on file as well. However,  we may want to stick to modifying the sourceMap field/object to ha e less possible collisions.

Comment (gzzz) [inline src/init.js]: Well, it is little complicated to me work with exports in v2.
In v1 things were simpler…

**Comment by gzzz** [comment]
https://github.com/gzzz/gulp-sourcemaps/commit/1ddace7b4f9ec431855729f11779631f5b9d993c

So, i've moved newline field into sourceMap object, but have found an issue with [gulp-uglify](https://github.com/terinjokes/gulp-uglify/).
If it is present in gulp pipe between sourcemaps init and write, [it recreates a sourceMap object](https://github.com/terinjokes/gulp-uglify/blob/master/minifier.js#L91) and drops newline field.

I think a ancillary file field – `sourcemapsNewline`, for example – is better way.

File: src/init.js (hunk 1 above)
Comment (gzzz) [inline src/init.js]: https://github.com/gzzz/gulp-sourcemaps/commit/1ddace7b4f9ec431855729f11779631f5b9d993c

So, i've moved newline field into sourceMap object, but have found an issue with [gulp-uglify](https://github.com/terinjokes/gulp-uglify/).
If it is present in gulp pipe between sourcemaps init and write, [it recreates a sourceMap object](https://github.com/terinjokes/gulp-uglify/blob/master/minifier.js#L91) and drops newline field.

I think a ancillary file field – `sourcemapsNewline`, for example – is better way.

Comment (nmccready) [inline src/init.js]: Good to know ok, so make this sourceMapExt field or gulpSourceMap because I will use the same object in another PR I am working on.

**Comment by nmccready** [comment]
You should be  PRin against 1.X the code is no different. It makes it
easier to roll the patch into both branches.

**Pull request closed without merging**